jaq-native/src/lib.rs

```diff
@@ -260,6 +269,69 @@ fn now() -> Result<f64, Error> {
 #[cfg(feature = "std")]
 const STD: &[(&str, usize, RunPtr)] = &[("now", 0, |_, _| box_once(now().map(Val::Float)))];
 
+#[cfg(feature = "math")]
+const MATH: &[(&str, usize, RunPtr)] = &[
+    math::f_f!(acos),
+    math::f_f!(acosh),
+    math::f_f!(asin),
+    math::f_f!(asinh),
+    math::f_f!(atan),
+    math::f_f!(atanh),
+    math::f_f!(cbrt),
+    math::f_f!(cos),
+    math::f_f!(cosh),
+    math::f_f!(erf),
+    math::f_f!(erfc),
+    math::f_f!(exp),
+    math::f_f!(exp10),
+    math::f_f!(exp2),
+    math::f_f!(expm1),
+    math::f_f!(fabs),
+    math::f_fi!(frexp),
+    math::f_f!("gamma", tgamma),
+    math::f_i!("ilogb_", ilogb),
+    math::f_f!(j0),
+    math::f_f!(j1),
+    math::f_f!(lgamma),
+    math::f_f!(log),
+    math::f_f!(log10),
+    math::f_f!(log1p),
+    math::f_f!(log2),
+    // logb is implemented in jaq-std
+    math::f_ff!(modf),
+    math::f_f!("nearbyint", round),
+    // pow10 is implemented in jaq-std
+    math::f_f!(rint),
+    // significand is implemented in jaq-std
+    math::f_f!(sin),
+    math::f_f!(sinh),
+    math::f_f!(sqrt),
+    math::f_f!(tan),
+    math::f_f!(tanh),
+    math::f_f!(tgamma),
+    math::f_f!(trunc),
+    math::f_f!(y0),
+    math::f_f!(y1),
+    math::ff_f!(atan2),
+    math::ff_f!(copysign),
+    // drem is implemented in jaq-std
+    math::ff_f!(fdim),
+    math::ff_f!(fmax),
+    math::ff_f!(fmin),
+    math::ff_f!(fmod),
+    math::ff_f!(hypot),
+    math::if_f!(jn),
+    math::fi_f!(ldexp),
+    math::ff_f!(nextafter),
+    math::ff_f!("nexttoward", nextafter),
+    math::ff_f!(pow),
+    math::ff_f!(remainder),
+    math::fi_f!("scalb", scalbn),
+    math::fi_f!("scalbln", scalbn),
```

Comment:
These two are not the same,  consider the following example:

~~~
$ jq -n 'scalb(1.5; 3.5), scalbln(1.5; 3.5)'
null
12
~~~

Reply:
That is surprising! Again, consider I just learned about many of these functions so I'm little more than an inefficient search engine at this point. Still, checking the man pages for both you get:

- [scalb](https://man7.org/linux/man-pages/man3/scalb.3.html):  `x * FLT_RADIX ** exp`
- [scalbln](https://man7.org/linux/man-pages/man3/scalbln.3.html): `x * FLT_RADIX ** exp`

My intuition is that one is an upgraded version of the other, with handling for special cases. However the example you gave when plugged into the expression supposed to match their definition gives (in say, plain jq) doesn't strike me as a special case at all:

```text
$ jq -n '1.5 * pow(2; 3.5)'
16.970562748477143
```

which is what gojq yields for `scalb(1.5; 3.5)`.

Reply:
I now see that (in the statically linked version of jq downloadable from the main site), the `scalb` function yields `nan` when the second argument is not an integer, while `scalbln` coerces the second argument to an integer.

```text
$ jq -n 'scalb(2.5; 2), scalbln(2.5; 2), scalbln(2.5; 2.99), (scalb(2.5; 2.99) | isnan)'
10
10
10
true
```

Reply:
I agree! And now everything clicked in my head 😆 the `ln` suffix stands for long! I kept assuming a logarithm was involved somewhere.